src/main/java/io/debezium/connector/vitess/connection/VitessReplicationConnection.java

```diff
@@ -99,6 +99,12 @@ public void onNext(Vtgate.VStreamResponse response) {
                     LOGGER.debug("VEvent: {}", vEvent);
                 }
 
+                // If we receive an empty transaction, there is nothing we can stream out
+                // to the downstream. Therefore, skip them directly here instead.
+                if (isEmptyTransaction(response)) {
```

Comment:
Skipping events like this look a very coarse handling to me. It conflicts with existing logic on event processing and skipping, which is mostly done within `messageDecoder.processMessage`. See https://github.com/debezium/debezium-connector-vitess/pull/73/files for how [BEGIN, COMMIT] events were handled. I highly suggestwe  can handle this case similarly to avoid trapdoor.